py4web/utils/form.py

```diff
@@ -202,6 +200,8 @@ def __init__(self,
                             self.vars[field.name] = value
                             if error:
                                 self.errors[field.name] = error
+                    if len(self.errors) == 0 and validation:
+                        self.errors = validation(self)
```

Comment:
No need to return errors given that validate can manipulate self.errors. Will change it after merging.